packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -9,13 +9,11 @@ import 'build_info.dart';
 import 'globals.dart';
 
 enum Artifact {
-  icudtlDat,
```

Comment:
On Android icudtl.dat is embedded within flutter.jar - it doesn't need to be referenced as an independent artifact.

We still download the icudtl.dat file for use by flutter_tester on the host, but flutter_tools doesn't need to access that.